pkg/controller/nodenetworkconfigurationpolicy/conditions/policy.go

```diff
@@ -0,0 +1,166 @@
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionProgressing,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+		"",
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+		"",
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionTrue,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured,
+		message,
+	)
+}
+
+func setPolicyNotMatching(conditions *nmstatev1alpha1.ConditionList, message string) {
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionProgressing,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode,
+		message,
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode,
+		message,
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionFalse,
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode,
+		message,
+	)
+}
+
+func setPolicyFailedToConfigure(conditions *nmstatev1alpha1.ConditionList, message string) {
+	setPolicyFailed(conditions, nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionFailedToConfigure, message)
+}
+
+func setPolicyFailed(conditions *nmstatev1alpha1.ConditionList, reason nmstatev1alpha1.ConditionReason, message string) {
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionDegraded,
+		corev1.ConditionTrue,
+		reason,
+		message,
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionAvailable,
+		corev1.ConditionFalse,
+		reason,
+		"",
+	)
+	conditions.Set(
+		nmstatev1alpha1.NodeNetworkConfigurationPolicyConditionProgressing,
+		corev1.ConditionFalse,
+		reason,
+		"",
+	)
+}
+
+func UpdatePolicyConditions(cli client.Client, policy *nmstatev1alpha1.NodeNetworkConfigurationPolicy) error {
+	logger := log.WithValues("policy", policy.Name)
+	// On conflict we need to re-retrieve enactments since the
+	// conflict can denote that the calculated policy conditions
+	// are now not accurate.
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		enactments := nmstatev1alpha1.NodeNetworkConfigurationEnactmentList{}
+		policyLabelFilter := client.MatchingLabels{nmstatev1alpha1.ENACTMENT_POLICY_LABEL: policy.Name}
+		err := cli.List(context.TODO(), &enactments, policyLabelFilter)
+		if err != nil {
+			return errors.Wrap(err, "geting enactments failed")
+		}
+
+		nodes := corev1.NodeList{}
+		nodeSelectorFilter := client.MatchingLabels(policy.Spec.NodeSelector)
```

Comment:
This scares me (and at the same time is super cool for what we have now). It works now because of all selectors we need are NodeSelectors... However, let's roll with this. We can address improving this once we have a reason (other selectors).

Just one question. What happens with `nil` NodeSelector?

Reply:
> A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic

From https://blog.golang.org/go-maps-in-action so we are good there if nil.